### PR TITLE
[MAINTENANCE] Minor clean-up

### PR DESCRIPTION
**Pull request by donaldheppner**
Changes proposed in this pull request:
- Removed some duplicate code and unused variables in validator and execution engine
- Added comments regarding our initial exploration of Batch Metric Store integration points

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [ ] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

File: great_expectations/execution_engine/execution_engine.py [hunk 1]
```diff
@@ -362,53 +362,39 @@ def resolve_metrics(
                     )
                 )
                 continue
+
             metric_fn_type = getattr(
                 metric_fn, "metric_fn_type", MetricFunctionTypes.VALUE
             )
-            if metric_fn_type in [
+
+            if not metric_fn_type in [
```
Comment (cdkini) [inline great_expectations/execution_engine/execution_engine.py]: Nitpick - could we do `metric_fn_type not in`?

File: great_expectations/validator/validator.py [hunk 2]
```diff
@@ -639,8 +639,6 @@ def _update_metric_value_kwargs_for_success_keys(
                 parameter_builder.metric_value_kwargs or {}
             )
 
-            key: str
-            value: Any
```
Comment (cdkini) [inline great_expectations/validator/validator.py:643]: So @alexsherstinsky tends to add these as a way of annotating the values in a comprehension - I would probably keep them but I'm okay either way.

File: great_expectations/validator/validator.py [hunk 3]
```diff
@@ -1119,7 +1108,7 @@ def resolve_validation_graph(
         exception_info: ExceptionInfo
 
         # noinspection SpellCheckingInspection
-        pbar = None
+        progress_bar = None
```
Comment (cdkini) [inline great_expectations/validator/validator.py:1117]: I think this was me 😬 

thank you for this!

**Review by cdkini — APPROVED** [review]

File: great_expectations/validator/validator.py [hunk 4]
```diff
@@ -703,14 +701,6 @@ def compute_metrics(
...
 
         resolved_metrics: Dict[Tuple[str, str, str], Any] = {}
-        # noinspection PyUnusedLocal
-        aborted_metrics_info: Dict[
-            Tuple[str, str, str],
-            Dict[str, Union[MetricConfiguration, List[Exception], int]],
-        ] = self.resolve_validation_graph(
-            graph=graph,
-            metrics=resolved_metrics,
-        )
```
Comment (anthonyburdi) [inline great_expectations/validator/validator.py]: I think we still need to run `self.resolve_validation_graph()` here even if not using the `aborted_metrics_info`. Instead you can replace `aborted_metrics_info` with `_` to throw away the value.

**Review by anthonyburdi — CHANGES_REQUESTED** [review]
Looks good aside from one requested change.

**Review by anthonyburdi — APPROVED** [review]
LGTM!

File: great_expectations/validator/validator.py (hunk 4 above)
Comment (donaldheppner) [reply great_expectations/validator/validator.py]: return value not required, but the call still is! good catch

**Pull request merged**